crates/ruma-events/src/room/message/audio.rs

```diff
@@ -90,6 +93,17 @@ impl AudioMessageEventContent {
     }
 }
 
+// Add support for captions via MSC2530
+impl Captionable for AudioMessageEventContent {
+    fn caption(&self) -> Option<&str> {
+        caption(&self.body, self.filename.as_deref())
+    }
+
+    fn formatted_caption(&self) -> Option<&FormattedBody> {
+        formatted_caption(&self.body, self.formatted.as_ref(), self.filename.as_deref())
+    }
+}
```

Comment:
This is almost what I meant - I meant removing the `Captionable` trait entirely and moving these into `impl AudioMessageEventContent` and such. Or is there some code you wrote downstream that wants to be generic over `Captionable` values?

Reply:
Ah ok, got it. Will update (and rename `captionable.rs` to `media_caption.rs`

Reply:
https://github.com/ruma/ruma/pull/1919/commits/749f5ca2f5053128db7422fbd4a11df729a60f7a